packages/terra-pills/src/PillList.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  KEY_SPACE,
+  KEY_RETURN,
+} from 'keycode-js';
+import PropTypes from 'prop-types';
+import classNamesBind from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import styles from './Pill.module.scss';
+import Pill from './Pill';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * The aria label string value for the pills container.
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * The content to be shown in the pills container. Should only contain the sub-component pills
+   */
+  children: PropTypes.node,
+  /**
+   * Determines if the Pill List rolls up or not.
+   */
+  isCollapsible: PropTypes.bool,
+};
+
+const defaultProps = {
+  isCollapsible: false,
+};
+
+const PillList = (props) => {
+  const {
+    ariaLabel,
+    children,
+    isCollapsible,
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const pillListRef = useRef();
+  const rollUpPillRef = useRef();
+  const [rollUpCount, setRollUpCount] = useState(0);
+  const [isRollUpRequired, setIsRollUpRequired] = useState(true);
+
+  const handleCollapse = () => {
+    let pillListHeight = pillListRef.current.getBoundingClientRect().height;
+    const pillHeight = rollUpPillRef.current.getBoundingClientRect().height + 12; // Margin + border width
```

Comment:
Can you obtain margin and border width should be obtained programmatically? Hardcoding it here will make the code brittle if styles change and as themes are applied.

Reply:
This line also caught my eye. I can see where you got the magic number of `12` from the Zeplin callouts, but this seems like something that needs to be accounted for in CSS.

Reply:
Addressed it here : https://github.com/cerner/terra-framework/pull/1260/commits/65bc8c8eb773f94746a47e8e3f29679e74fef500
Changed the logic to retrieve the margins programatically from the computed styles.